tests/Microsoft.Identity.Test.Unit/CoreTests/RegionDiscoveryProviderTests.cs

```diff
@@ -137,6 +138,20 @@ public async Task ResponseFromUserProvidedRegionDifferentFromRegionDetectedAsync
             regionDiscoveryProvider.Clear();
         }
 
+        [TestMethod]
+        public async Task SuccessfulResponseFromRegionalizedAuthorityAsync()
```

Comment:
@henrik-me This is the test method that tests the scenario. Calling `TryGetMetadataAsync` twice because it gets called twice in the end-to-end flow.

Reply:
I thought the idea was that someone calls aquiretokenforclient 2 times and gets into this situation? Or are you saying that our code does that before returning the token? .. in which case why is that?

Reply:
The latter - our code does it. Not sure why, investigation is needed.